hare/broker.go

```diff
@@ -327,124 +267,61 @@ func (b *Broker) HandleEligibility(ctx context.Context, em *types.HareEligibilit
 	return false
 }
 
-func (b *Broker) getInbox(id types.LayerID) chan any {
-	b.mu.RLock()
-	defer b.mu.RUnlock()
-	out, exist := b.outbox[id.Uint32()]
-	if !exist {
-		return nil
-	}
-	return out
-}
-
-func (b *Broker) handleEarlyMessage(logger log.Log, layer types.LayerID, nodeID types.NodeID, msg any) error {
+// Register a layer to receive messages
+// Note: the registering instance is assumed to be started and accepting messages.
+func (b *Broker) Register(ctx context.Context, id types.LayerID) (chan any,
+	error,
+) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
-	layerNum := layer.Uint32()
-	if _, exist := b.pending[layerNum]; !exist { // create buffer if first msg
-		b.pending[layerNum] = make([]any, 0, inboxCapacity)
+	if !id.After(b.latestLayer) { // should expect to update only newer layers
+		b.WithContext(ctx).With().Error("tried to update a previous layer",
+			log.Stringer("this_layer", id),
```

Comment:
i think it's better to simply use the default field name (i.e just put `id` here)

when debugging, it's useful to search all the loki log where layer_id == blah

Reply:
ideally if this happen we should log.Fatal this to signal a programming error. 
at the very least it should not hide this programming error and move on.
should log.Error AND return an error.

best to add test for this case.

Reply:
So in this case I was maintaining the previous logic that existed in `setLatestLayer`, if we were to return an error here that would mean that once a a process has registered for a layer, you can only register for later layers. Is this the case, or are there situations where you could end up wanting to register a lower layer? What about if tortoise re-orgs and the new chain is at a lower layer?

Reply:
no. hare only run one layer at most once. so it SHOULD never go back in time to register an older layer.
the only possibility is that the `Register` is called on a separate goroutine, but that's not the case now.

Reply:
Ok cool, will update to return an error here :+1:

Reply:
Actually I'm going to go for a panic in this case because it seems like this would be a coding error if it occurred.